substrate/frame/identity/src/lib.rs

```diff
@@ -1377,34 +1377,49 @@ impl<T: Config> Pallet<T> {
 		target: &T::AccountId,
 	) -> Result<(BalanceOf<T>, BalanceOf<T>), DispatchError> {
 		// Identity Deposit
-		let new_id_deposit = IdentityOf::<T>::try_mutate(
-			&target,
-			|identity_of| -> Result<BalanceOf<T>, DispatchError> {
-				let (reg, _) = identity_of.as_mut().ok_or(Error::<T>::NoIdentity)?;
-				// Calculate what deposit should be
-				let encoded_byte_size = reg.info.encoded_size() as u32;
-				let byte_deposit =
-					T::ByteDeposit::get().saturating_mul(<BalanceOf<T>>::from(encoded_byte_size));
-				let new_id_deposit = T::BasicDeposit::get().saturating_add(byte_deposit);
-
-				// Update account
-				Self::rejig_deposit(&target, reg.deposit, new_id_deposit)?;
-
-				reg.deposit = new_id_deposit;
-				Ok(new_id_deposit)
-			},
-		)?;
-
-		// Subs Deposit
-		let new_subs_deposit = SubsOf::<T>::try_mutate(
-			&target,
-			|(current_subs_deposit, subs_of)| -> Result<BalanceOf<T>, DispatchError> {
-				let new_subs_deposit = Self::subs_deposit(subs_of.len() as u32);
-				Self::rejig_deposit(&target, *current_subs_deposit, new_subs_deposit)?;
-				*current_subs_deposit = new_subs_deposit;
-				Ok(new_subs_deposit)
-			},
-		)?;
+		let has_identity = IdentityOf::<T>::contains_key(&target);
+		// This should really always be true, but don't insert random storage on the off chance it's
+		// not.
+		let new_id_deposit = if has_identity {
```

Comment:
This is redundant code because if there is no identity, the function will fail on `let (reg, _) = identity_of.as_mut().ok_or(Error::<T>::NoIdentity)?;`. It works with `IdentityOf` but not with `SubsOf` because `IdentityOf` is `OptionQuery` but `SubsOf` is `ValueQuery`. Guarding the subs rejig with a `SubsOf::<T>::contains_key(&target)` is enough IMO.

Reply:
Ah yeah, you are right.